app/services/streaming/streaming.ts

```diff
@@ -61,43 +71,346 @@ interface IOBSOutputSignalInfo {
+  @Inject() private streamSettingsService: StreamSettingsService;
+  @Inject() private outputSettingsService: OutputSettingsService;
+  @Inject() private windowsService: WindowsService;
+  @Inject() private usageStatisticsService: UsageStatisticsService;
+  @Inject() private notificationsService: NotificationsService;
+  @Inject() private userService: UserService;
+  @Inject() private incrementalRolloutService: IncrementalRolloutService;
   @Inject() private videoEncodingOptimizationService: VideoEncodingOptimizationService;
   @Inject() private navigationService: NavigationService;
   @Inject() private customizationService: CustomizationService;
   @Inject() private restreamService: RestreamService;
+  @Inject() private hostsService: HostsService;
+  @Inject() private facebookService: FacebookService;
+  @Inject() private youtubeService: YoutubeService;
+  @Inject() private twitterService: TwitterService;
 
   streamingStatusChange = new Subject<EStreamingState>();
   recordingStatusChange = new Subject<ERecordingState>();
   replayBufferStatusChange = new Subject<EReplayBufferState>();
   replayBufferFileWrite = new Subject<string>();
+  streamInfoChanged = new Subject<StreamInfoView>();
 
   // Dummy subscription for stream deck
   streamingStateChange = new Subject<void>();
 
   powerSaveId: number;
 
-  static initialState = {
+  private resolveStartStreaming: Function = () => {};
+  private rejectStartStreaming: Function = () => {};
+
+  static initialState: IStreamingServiceState = {
     streamingStatus: EStreamingState.Offline,
     streamingStatusTime: new Date().toISOString(),
     recordingStatus: ERecordingState.Offline,
     recordingStatusTime: new Date().toISOString(),
     replayBufferStatus: EReplayBufferState.Offline,
     replayBufferStatusTime: new Date().toISOString(),
     selectiveRecording: false,
+    info: {
+      lifecycle: 'empty',
+      error: null,
+      checklist: {
+        applyOptimizedSettings: 'not-started',
+        twitch: 'not-started',
+        youtube: 'not-started',
+        mixer: 'not-started',
+        facebook: 'not-started',
+        setupRestream: 'not-started',
+        startVideoTransmission: 'not-started',
+        publishYoutubeBroadcast: 'not-started',
+        postTweet: 'not-started',
+      },
+    },
   };
 
   init() {
     obs.NodeObs.OBS_service_connectOutputSignals((info: IOBSOutputSignalInfo) => {
       this.handleOBSOutputSignal(info);
     });
+
+    // watch for StreamInfoView at emit `streamInfoChanged` event if something has been hanged there
+    this.store.watch(
+      () => {
+        this.views.chatUrl; // read `chatUrl` to tell vuex that this computed property is reactive
+        return this.views;
+      },
+      val => {
+        // show the error if child window is closed
+        if (val.info.error && !this.windowsService.state.child.isShown) {
+          this.showGoLiveWindow();
+        }
+        this.streamInfoChanged.next(val);
```

Comment:
Hooking up rxjs events to automatically dispatch on Vuex store changes is probably something work encapsulating as a proper pattern with a helper in the future.  This seems like a common use case in our code and we often update the subject manually.  Worth thinking about.  No action on this PR.